src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/service/haflow/HaFlowUpdateService.java

```diff
@@ -96,6 +136,230 @@ public void handleUpdateRequest(
+        return new FlowEndpoint(switchId, port, vlan, innerVlan);
+    }
+
+    @VisibleForTesting
+    void updateHaFlowPartially(HaFlow target, HaFlowPartialUpdateRequest source) {
+        Optional.ofNullable(source.getMaxLatency()).ifPresent(target::setMaxLatency);
+        Optional.ofNullable(source.getMaxLatencyTier2()).ifPresent(target::setMaxLatencyTier2);
+        Optional.ofNullable(source.getPeriodicPings()).ifPresent(target::setPeriodicPings);
+        Optional.ofNullable(source.getPinned()).ifPresent(target::setPinned);
+        Optional.ofNullable(source.getPriority()).ifPresent(target::setPriority);
+        Optional.ofNullable(source.getStrictBandwidth()).ifPresent(target::setStrictBandwidth);
+        Optional.ofNullable(source.getDescription()).ifPresent(target::setDescription);
+        if (source.getSubFlows() != null) {
+            for (HaSubFlowPartialUpdateDto subFlow : source.getSubFlows()) {
+                if (subFlow != null) {
+                    Optional.ofNullable(subFlow.getDescription())
+                            .ifPresent(description -> target.getHaSubFlowOrThrowException(
+                                    subFlow.getFlowId()).setDescription(description));
+                }
+            }
+        }
+    }
+
+    private void validateSubFlowIds(HaFlowPartialUpdateRequest request, String haFlowId, Set<String> existingSubFlowIds)
+            throws FlowProcessingException {
+        try {
+            haFlowValidator.validateHaSubFlowIds(haFlowId, existingSubFlowIds, request.getSubFlows());
+        } catch (InvalidFlowException e) {
+            throw new FlowProcessingException(e.getType(), e.getMessage());
+        }
+    }
+
+    @VisibleForTesting
+    boolean isFullUpdateRequired(HaFlowPartialUpdateRequest request, HaFlow haFlow) {
+        return isUpdateRequiredByPathComputationStrategy(request, haFlow)
+                || isUpdateRequiredBySharedEndpoint(request, haFlow)
+                || isUpdateRequiredBySubFlowEndpoints(request, haFlow)
+                || isFieldUpdated(request.getMaximumBandwidth(), haFlow.getMaximumBandwidth())
+                || isFieldUpdated(request.getAllocateProtectedPath(), haFlow.isAllocateProtectedPath())
+                || isFieldUpdated(request.getIgnoreBandwidth(), haFlow.isIgnoreBandwidth())
+                || isFieldUpdated(request.getEncapsulationType(), haFlow.getEncapsulationType())
+                || isUpdateRequiredByDiverseFlowIdField(request, haFlow);
+    }
+
+    private static boolean isUpdateRequiredByPathComputationStrategy(
+            HaFlowPartialUpdateRequest request, HaFlow haFlow) {
+        PathComputationStrategy strategy = Optional.ofNullable(request.getPathComputationStrategy())
+                .orElse(haFlow.getPathComputationStrategy());
+        boolean strategyIsLatencyBased = LATENCY_BASED_STRATEGIES.contains(strategy);
+        boolean changedMaxLatency = isFieldUpdated(request.getMaxLatency(), haFlow.getMaxLatency());
```

Comment:
minor, non-blocking: the value on 293 is only needed if 292 is true and 295 is false, but it is calculated every time even when we know that the result will be discarded. I'd suggest to split/extract it in separate methods and call them only when needed (for example, as an operand in `||`)

Reply:
good point